api/src/routes/data-ingestion-router.ts

```diff
@@ -0,0 +1,62 @@
+import express, { Request, Response } from "express";
+
+import { db } from "@/data/db-client";
+import { RequireAdmin } from "../middleware";
+import {
+  CreateService as DataIngestionCreateService,
+  DataIngestionSourceService,
+  UserService,
+} from "@/services";
+import { isNil } from "lodash";
+
+export const dataIngestionRouter = express.Router();
+
+dataIngestionRouter.get("/", async (req: Request, res: Response) => {
+  const list = await db("data_ingestions").orderBy("source_id");
+  return res.json({ dataIngestions: list, totalCount: list.length });
+});
+
+dataIngestionRouter.post("/", async (req: Request, res: Response) => {
+  const users = new UserService();
+  const dataIngestionSource = new DataIngestionSourceService();
+  const { source_id, user_id } = req.body;
+
+  if (isNil(req.files?.csvFile)) return res.status(400).json({ error: "Missing file" });
+  if (isNil(user_id)) return res.status(422).json({ error: "Missing user_id" });
+  if (isNil(source_id)) return res.status(422).json({ error: "Missing source_id" });
+
+  if (isNil(users.getById(Number(user_id)))) {
+    return res.status(422).json({ error: "Invalid user_id" });
+  }
+
+  if (isNil(dataIngestionSource.getById(Number(source_id)))) {
+    return res.status(422).json({ error: "Invalid source_id" });
+  }
+
+  const uploaded = Array.isArray(req.files.csvFile) ? req.files.csvFile[0] : req.files.csvFile;
+
+  try {
+    await DataIngestionCreateService.perform(uploaded.data, Number(source_id), Number(user_id));
+    return res.json({ success: true });
+  } catch (err: any) {
+    console.error(" DataIngestionService Error:", err);
+    return res.status(500).json({ error: err.message });
```

Comment:
Prefer this pattern for errors:
```ts
console.error(`Assessment deletion failed: ${error}`, { error })
return this.response.status(422).json({ 
  message: `Assessment deletion failed: ${error}` 
})
```

Generally is use 422 for POST|PATCH|DELETE, and 400 for GET.
Generally I leave 500 errors for the server being kill or something.
See https://developer.mozilla.org/en-US/docs/Web/HTTP/Status